### ISSUE TITLE
[@types/node] address() definition in http module

### ISSUE THREAD

**Issue by mvasin** [DefinitelyTyped/DefinitelyTyped#31955]
I've noticed that `address()` definition in `http` (and maybe related modules) is off.

Here's the definition:
```
address(): AddressInfo | string;
```

Here's the code that proves it wrong:
```
const http = require('http');
const server = http.createServer();
server.address() === null
// => true
```

So `address(): AddressInfo | null` could be more correct.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @sheetalkamat

**Comment by mvasin** [comment]
I've mentioned based on git blame, but issue template suggests I should mention according to `Definitions by` of `types/node`, so here we go: @Microsoft @DefinitelyTyped @parambirs @tellnes @WilcoBakker @octo-sniffle @smac89 @Flarna @mwiktorczyk @wwwy3y3 @DeividasBakanas @kjin @alvis @eps1lon @Hannes-Magnusson-CK @jkomyno @ajafff @hoo29 @n-e @brunoscheufler @mohsen1 @KSXGitHub @a-tarasyuk @islishude @r3nya @eyqs

**Comment by Flarna** [comment]
I think it should be `AddressInfo | string | null`.
If the server listens on an unix domain socket or windows pipe you get the path as string.
Feel free to create a PR.

**Comment by mvasin** [comment]
OK, I'll come up with a PR.

**Comment by mvasin** [comment]
I opened a WIP PR #31985. Please have a look.

**Comment by Flarna** [comment]
@mvasin This seems to be fixed via #32878. Could you please check and close this issue if it works for you?

**Comment by mvasin** [comment]
@Flarna Indeed, seems to be fixed.
